projects/plugins/wpcomsh/performance-profiler/performance-profiler.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * Adds the generated performance report url for each page and also in v1 site settings payload for homepage.
+ *
+ * @package performance-profiler
+ */
+
+/**
+ * Adds the generated performance report url for each page.
+ */
+function wpcom_performance_url_rest_field() {
+	register_rest_field(
+		'page',
+		'wpcom_performance_report_url',
+		array(
+			'get_callback'    => 'wpcom_performance_report_url_get_value',
+			'update_callback' => 'wpcom_performance_report_url_update_value',
+			'schema'          => array(
+				'type'        => 'string',
+				'description' => __( 'The performance report URL for this page.', 'wpcomsh' ),
+				'default'     => '',
+			),
+		)
+	);
+}
+
+/**
+ * Filter the performance report URL field.
+ *
+ * @param WP_REST_Response $response The response object.
+ * @param WP_Post          $post     The post object.
+ * @param WP_REST_Request  $request  The request object.
+ * @return WP_REST_Response
+ */
+function filter_wpcom_performance_report_url( $response, $post, $request ) {
+	$fields = $request->get_param( '_fields' );
+
+	if ( is_string( $fields ) ) {
+		$fields = explode( ',', $fields );
+	}
+
+	if ( ! is_array( $fields ) ) {
+		$fields = array();
+	}
+
+	if ( empty( $fields ) || ! in_array( 'wpcom_performance_report_url', $fields, true ) ) {
+		unset( $response->data['wpcom_performance_report_url'] );
+	}
+
+	return $response;
+}
+
+add_filter( 'rest_prepare_page', 'filter_wpcom_performance_report_url', 10, 3 );
```

Comment:
It's too bad that this filter only removes the field _after_ we've already wasted time doing the DB query.

After seeing all the extra fields that already get returned by default (e.g. all the Jetpack fields), maybe it's not such a big deal to do the default thing and return our extra field by default 🤔

Reply:
yeah jetpack returns a bunch of default fields. I don't have a problem filter fields, we are only asking for few fields on the frontend, at least in makes the http call/response faster